packages/@ember/-internals/glimmer/lib/components/link-to.ts

```diff
@@ -80,6 +80,11 @@ class LinkTo extends InternalComponent implements DeprecatingInternalComponent {
       !('model' in this.args.named && 'models' in this.args.named)
     );
 
+    assert(
+      'Passing the `@tagName` argument to <LinkTo> is not supported.',
+      !('tagName' in this.args.named)
+    );
```

Comment:
Two concerns:

With this implemented as an assert, what does that mean for production? Would a `@tagName` be applied in production, even if it the assert stopped you in development? Seems bad if so. Maybe it should be a `throw`.

But, further, IMO we don't need a check at all. I don't think we're planning to add asserts for every old component argument which was removed in 4.0. They were removed, which means passing one of those arguments (like `@tagName`) could just mean it is ignored. I can ask around on this second point, but the first concern about production behavior *not* applying the tag name needs to be confirmed.

Reply:
Yup, makes sense re: `assert`. As to the second point, classic components will keep `tagName`, so it seems like something special will need to be done for `linkTo`.